### PR TITLE
Bulk Indexing for S2 NRT into Sandbox Database

### PR DESCRIPTION
**Pull request by kieranricardo**
This DAG is basically a copy of the `odc` DAG, but with the database name and DAG name changed.

### CODE REVIEW

**Comment by pindge** [comment]
No, u cannot do that for NRT

File: dags/sentinel_2_nrt/sentinel_2_nrt_batch_indexing_sandbox.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    "retry_delay": timedelta(minutes=5),
+    "env_vars": {
+        # TODO: Pass these via templated params in DAG Run
+        "DB_HOSTNAME": DB_HOSTNAME,
+        "DB_DATABASE": "sandbox",
+        "DB_PORT": "5432",
+    },
+    # Lift secrets into environment variables
+    "secrets": [
+        Secret("env", "DB_USERNAME", SECRET_ODC_WRITER_NAME, "postgres-username"),
```
Comment (pindge) [inline dags/sentinel_2_nrt/sentinel_2_nrt_batch_indexing_sandbox.py:47]: also sandbox db has a different user

File: dags/sentinel_2_nrt/sentinel_2_nrt_batch_indexing_sandbox.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        Secret(
+            "env", "AWS_SECRET_ACCESS_KEY", SECRET_AWS_NAME, "AWS_SECRET_ACCESS_KEY"
+        ),
+    ],
+}
+
+uri_list = []
+for n in range(1, 8):  # cover an entire week
+    uri_date = (datetime.today() - timedelta(days=n)).strftime("%Y-%m-%d")
+    URI = f"s3://dea-public-data/L2/sentinel-2-nrt/S2MSIARD/{uri_date}/**/ARD-METADATA.yaml"
```
Comment (pindge) [inline dags/sentinel_2_nrt/sentinel_2_nrt_batch_indexing_sandbox.py:60]: u should watch my video.

**Review by pindge — CHANGES_REQUESTED** [review]

File: dags/sentinel_2_nrt/sentinel_2_nrt_batch_indexing_sandbox.py (hunk 1 above)
Comment (kieranricardo) [reply dags/sentinel_2_nrt/sentinel_2_nrt_batch_indexing_sandbox.py:47]: just tested and `sandbox` has a user w/ the same name and password so I think this works

**Comment by kieranricardo** [comment]
@pindge I just checked and `sandbox` DB does have the `eo_plus` metadata type, so I think this DAG will actually work

File: dags/sentinel_2_nrt/sentinel_2_nrt_batch_indexing_sandbox.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+from infra.variables import (
+    DB_DATABASE,
+    DB_HOSTNAME,
+    SECRET_ODC_WRITER_NAME,
+    SECRET_AWS_NAME,
+)
+from sentinel_2_nrt.env_cfg import (
+    INDEXING_PRODUCTS,
+)
+from infra.podconfig import NODE_AFFINITY
```
Comment (pindge) [inline dags/sentinel_2_nrt/sentinel_2_nrt_batch_indexing_sandbox.py:27]: This has been changed to `ONDEMAND_NODE_AFFINITY`

File: dags/sentinel_2_nrt/sentinel_2_nrt_batch_indexing_sandbox.py [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+# DAG CONFIGURATION
+DEFAULT_ARGS = {
+    "owner": "Pin Jin",
```
Comment (pindge) [inline dags/sentinel_2_nrt/sentinel_2_nrt_batch_indexing_sandbox.py:31]: Change this please

**Review by pindge — CHANGES_REQUESTED** [review]

File: dags/sentinel_2_nrt/sentinel_2_nrt_batch_indexing_sandbox.py (hunk 4 above)
Comment (kieranricardo) [reply dags/sentinel_2_nrt/sentinel_2_nrt_batch_indexing_sandbox.py]: whoops sorry! didn't mean to shove the blame onto you

File: dags/sentinel_2_nrt/sentinel_2_nrt_batch_indexing_sandbox.py [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+from airflow.kubernetes.secret import Secret
+from airflow.contrib.operators.kubernetes_pod_operator import KubernetesPodOperator
+
+from textwrap import dedent
+
+from infra.images import INDEXER_IMAGE
+from infra.variables import (
+    DB_DATABASE,
+    DB_HOSTNAME,
+    SECRET_ODC_WRITER_NAME,
```
Comment (pindge) [inline dags/sentinel_2_nrt/sentinel_2_nrt_batch_indexing_sandbox.py:21]: no this one wont work.

File: dags/sentinel_2_nrt/sentinel_2_nrt_batch_indexing_sandbox.py (hunk 1 above)
Comment (pindge) [reply dags/sentinel_2_nrt/sentinel_2_nrt_batch_indexing_sandbox.py:47]: create a sandbox db secret in infra variables

```
Variables.get('sandbox-db-writter-secret', 'ows-db')

**Review by pindge — CHANGES_REQUESTED** [review]

File: dags/sentinel_2_nrt/sentinel_2_nrt_batch_indexing_sandbox.py [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+    "start_date": datetime(2020, 6, 14),
+    "email": ["pin.jin@ga.gov.au"],
+    "email_on_failure": False,
+    "email_on_retry": False,
+    "retries": 1,
+    "retry_delay": timedelta(minutes=5),
+    "env_vars": {
+        # TODO: Pass these via templated params in DAG Run
+        "DB_HOSTNAME": DB_HOSTNAME,
+        "DB_DATABASE": "sandbox",
```
Comment (pindge) [inline dags/sentinel_2_nrt/sentinel_2_nrt_batch_indexing_sandbox.py:42]: move this down to secret, 

for example
```
 Secret("env", "DB_PASSWORD", "odc-writer", "postgres-password"),
```

**Review by pindge — APPROVED** [review]

**Comment by kieranricardo** [comment]
@pindge I think I fixed all the things we talked about, can you have another look please?

**Comment by pindge** [comment]
> @pindge I think I fixed all the things we talked about, can you have another look please?

can u fix the conflict?

**Pull request closed without merging**